pageserver/src/metrics.rs

```diff
@@ -491,10 +491,87 @@ pub fn remove_tenant_metrics(tenant_id: &TenantId) {
 
 use futures::Future;
 use pin_project_lite::pin_project;
+use std::collections::HashMap;
 use std::pin::Pin;
+use std::sync::{Arc, Mutex};
 use std::task::{Context, Poll};
 use std::time::Instant;
 
+#[derive(Default)]
```

Comment:
This one is wrong. We need to set `tenant_id` and `timeline_id`. Will post a fix